src/core/public/doc_links/doc_links_service.ts

```diff
@@ -39,10 +40,33 @@ interface StartDeps {
 export class DocLinksService {
   public setup() {}
   public start({ injectedMetadata }: StartDeps): DocLinksStart {
-    const DOC_LINK_VERSION =
-      injectedMetadata.getOpenSearchDashboardsBranch() === 'main'
-        ? 'latest'
-        : injectedMetadata.getOpenSearchDashboardsBranch();
+    const buildVersion = injectedMetadata.getOpenSearchDashboardsVersion();
+    const pkgBranch = injectedMetadata.getOpenSearchDashboardsBranch();
+    /**
+     * OpenSearch server uses the `branch` property from `package.json` to
+     * build links to the documentation. If set to `main`, it would use `/latest`
+     * and if not, it would use the `version` to construct URLs.
+     */
+    let branch;
+    if (pkgBranch === 'main') {
+      branch = 'latest';
+    } else {
+      const parsedBuildVersion = parse(buildVersion);
+      if (parsedBuildVersion) {
+        branch = `${parsedBuildVersion.major}.${parsedBuildVersion.minor}`;
+      } else {
+        const validDocPathsPattern = /^\d+\.\d+$/;
+        if (validDocPathsPattern.test(pkgBranch)) {
+          branch = pkgBranch;
+        } else {
+          // package version was not parsable and branch is unusable
+          throw new Error(
```

Comment:
![image](https://user-images.githubusercontent.com/13493605/226507233-b12da14e-4bdf-47e5-b3e1-f5ab88345f0b.png)
When the logic goes to here, the dashboard will display such error page. But it won't be here because dashboard will check the version with OpenSearch core and thus it will always be parsed successfully by `semver`.